sunpy/map/mapbase.py

```diff
@@ -2730,8 +2730,9 @@ def contour(self, level, **kwargs):
             # find_contours expects a scalar below
             level = level[0]
 
-        contours = measure.find_contours(self.data, level=level, **kwargs)
-        contours = [self.wcs.array_index_to_world(c[:, 0], c[:, 1]) for c in contours]
+        gen = contour_generator(z=self.data,name="threaded", **kwargs)
```

Comment:
```suggestion
        gen = contour_generator(z=self.data, name="threaded", **kwargs)
```
I think we should instead check for name in kwargs. If a user passess in their own name, we want to check that it correctly is set.